modules/gke/main.tf

```diff
@@ -109,3 +113,216 @@ resource "google_service_account_iam_binding" "workload_identity" {
+  ]
+}
+
+# Create a ConfigMap containing the disk setup script
+resource "kubernetes_config_map" "disk_setup_script" {
+  count = var.enable_disk_setup ? 1 : 0
+
+  metadata {
+    name      = "disk-setup-script"
+    namespace = kubernetes_namespace.disk_setup[0].metadata[0].name
+  }
+
+  data = {
+    "bootstrap.sh" = file("${path.module}/bootstrap.sh")
+  }
+
+  depends_on = [
+    kubernetes_namespace.disk_setup
+  ]
+}
+
+resource "kubernetes_daemonset" "disk_setup" {
+  count = var.enable_disk_setup ? 1 : 0
+
+  metadata {
+    name      = "disk-setup"
+    namespace = kubernetes_namespace.disk_setup[0].metadata[0].name
+    labels = {
+      "app.kubernetes.io/managed-by" = "terraform"
+      "app.kubernetes.io/part-of"    = "materialize"
+      "app"                          = "disk-setup"
+    }
+  }
+
+  spec {
+    selector {
+      match_labels = {
+        app = "disk-setup"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "disk-setup"
+        }
+      }
+
+      spec {
+        security_context {
+          run_as_non_root = false
+          run_as_user     = 0
+          fs_group        = 0
+          seccomp_profile {
+            type = "RuntimeDefault"
+          }
+        }
+
+        affinity {
+          node_affinity {
+            required_during_scheduling_ignored_during_execution {
+              node_selector_term {
+                match_expressions {
+                  key      = "materialize.cloud/disk"
+                  operator = "In"
+                  values   = ["true"]
+                }
+              }
+            }
+          }
+        }
+
+        # Use host network and PID namespace
+        host_network = true
+        host_pid     = true
+
+        init_container {
+          name  = "disk-setup"
+          image = "debian:bullseye-20250407-slim"
+
+          command = ["/bin/bash", "/scripts/bootstrap.sh"]
+
+          resources {
+            limits = {
+              cpu    = "200m"
+              memory = "256Mi"
+            }
+            requests = {
+              cpu    = "100m"
+              memory = "128Mi"
+            }
+          }
+
+          security_context {
+            privileged  = true
+            run_as_user = 0
+          }
+
+          env {
+            name  = "SCRIPT_VERBOSE"
+            value = "true"
+          }
+
+          # Mount all necessary host paths
+          volume_mount {
+            name       = "scripts"
+            mount_path = "/scripts"
+          }
+
+          volume_mount {
+            name       = "mnt"
+            mount_path = "/mnt"
+          }
+        }
+
+        container {
+          name  = "pause"
+          image = "gcr.io/google_containers/pause:3.2"
+
+          resources {
+            limits = {
+              cpu    = "50m"
+              memory = "64Mi"
+            }
+            requests = {
+              cpu    = "10m"
+              memory = "32Mi"
```

Comment:
See above.

These can also likely drop to almost nothing, as the pause container is literally doing nothing.
Maybe 1m CPU and 8Mi memory?